webapp/src/main/java/uk/gov/dvsa/motr/web/resource/VrmResource.java

```
@@ -17,74 +24,127 @@
+    ) {
+
+        this.baseUrl = baseUrl;
+        this.motrSession = motrSession;
         this.renderer = renderer;
         this.client = client;
     }
 
     @GET
     public String vrmPageGet() throws Exception {
 
-        return renderer.render("vrm", emptyMap());
+        String regNumber = this.motrSession.getRegNumberFromSession();
+
+        Map<String, Object> modelMap = new HashMap<>();
+        updateMapBasedOnReviewFlow(modelMap);
+
+        modelMap.put(VRM_MODEL_KEY, regNumber);
+
+        if (this.motrSession.visitingFromReviewPage()) {
+            modelMap.put("back_location", "review");
+        }
+
+        return renderer.render("vrm", modelMap);
     }
 
     @POST
-    public String vrmPagePost(@FormParam("regNumber") String formParamVrm) {
+    public Response vrmPagePost(@FormParam("regNumber") String formParamVrm) throws Exception {
 
         String vrm = normalizeFormInputVrm(formParamVrm);
 
         Map<String, Object> modelMap = new HashMap<>();
+        updateMapBasedOnReviewFlow(modelMap);
+
         modelMap.put(VRM_MODEL_KEY, vrm);
         modelMap.put(SHOW_INLINE_KEY, true);
 
         VrmValidator validator = new VrmValidator();
-
         if (validator.isValid(vrm)) {
 
-            try {
+            this.motrSession.setVrm(vrm);
 
+            try {
                 Optional<VehicleDetails> vehicle = this.client.fetch(vrm);
                 if (!vehicle.isPresent()) {
                     modelMap.put(MESSAGE_KEY, VEHICLE_NOT_FOUND_MESSAGE);
                     modelMap.put(SHOW_INLINE_KEY, false);
-                }
+                } else {
 
+                    if (this.motrSession.visitingFromReviewPage()) {
+                        return Response.seeOther(getFullUriForPage("review")).build();
+                    }
+
+                    return Response.seeOther(getFullUriForPage("email")).build();
+                }
             } catch (VehicleDetailsClientException exception) {
```

at least rethrow. Never suppress any exceptions unless you know they will not happen.